lrlex/src/lib/parser.rs

```diff
@@ -361,9 +365,68 @@ impl<StorageT: TryFrom<usize>> LexParser<StorageT> {
         &self,
         off: usize,
         re_str: &'a str,
-    ) -> LexInternalBuildResult<(Vec<usize>, &'a str)> {
+    ) -> LexInternalBuildResult<(Vec<usize>, std::borrow::Cow<'a, str>)> {
         if !re_str.starts_with('<') {
-            Ok((vec![], re_str))
+            /// This implements the 'Table: Escape Sequences in lex' from POSIX lex specification
+            ///
+            /// Most of the escape handling is left to regex, except this part:
+            ///
+            /// Escape: \c
+            ///
+            /// Description:
+            /// A <backslash> character followed by any character not described in this table or in the table in
+            /// XBD File Format Notation ( '\\', '\a', '\b', '\f' , '\n', '\r', '\t', '\v' ).
+            ///
+            /// Meaning: The character 'c', unchanged.
+            fn unescape(re: Cow<str>) -> Cow<str> {
+                let re_str: &str = re.borrow();
+                let mut re_chars = re_str.char_indices().peekable();
+                let mut cursor = loop {
+                    if let Some((i, c)) = re_chars.next() {
+                        if c == '\\' {
+                            if let Some((j, p)) = re_chars.peek() {
+                                let s = &re_str[*j..*j + p.len_utf8()];
+                                if !(RE_ERE_QUOTED_CHARS.is_match(s) || RE_LEX_ESC.is_match(s)) {
+                                    break Some((i, s, j));
+                                }
+                            }
+                        }
+                    } else {
+                        break None;
+                    }
+                };
+                if cursor.is_none() {
+                    return re;
+                }
+                let mut unescaped = re_str.to_owned();
+                let mut diff = 0usize;
+                'outer: loop {
+                    if let Some((i, s, j)) = cursor {
+                        if !(RE_ERE_QUOTED_CHARS.is_match(s) || RE_LEX_ESC.is_match(s)) {
+                            unescaped.replace_range(i - diff..*j - diff, "");
```

Comment:
I guess it'd be more efficient to use `with_capacity` to allocate `unescaped` above, copying up to `cursor`, and then instead of `replace_range`, call `push_str` here.

Reply:
Should be fixed in [cf5b0b1](https://github.com/softdevteam/grmtools/pull/333/commits/cf5b0b177f76defff3a0afd47295226af0fc444e)